vcGL/src/vcTextureHelper.cpp

```diff
@@ -343,21 +364,26 @@ udResult vcTexture_SaveImage(vcTexture *pTexture, vcFramebuffer *pFramebuffer, c
   UD_ERROR_NULL(pFrameBufferPixels, udR_MemoryAllocationFailure);
 
   vcTexture_BeginReadPixels(pTexture, 0, 0, currSize.x, currSize.y, pFrameBufferPixels, pFramebuffer);
-
   UD_ERROR_CHECK(vcTexture_ConvertPixels(pFrameBufferPixels, format, &pPixelsOut, vcTextureFormat_RGBA8, currSize.x * currSize.y));
+  
+  if (udStrEndsWithi(pFilename, "png"))
+    stbiResult = stbi_write_png_to_func(vcTexture_StbCallback, &callbackResult, (int)currSize.x, (int)currSize.y, (int)pixelSize, pPixelsOut, 0);
```

Comment:
Just do the `UD_ERROR_IF` when you call, storing the `stbiResult` is unnecessary. Also, `pixelSize` is wrong, it needs to be `4`, that's why it was `4`.